IPs/RSKIP146.md

```diff
@@ -0,0 +1,144 @@
+
+### Events
+
+#### Lock - _new event_
+
+A new event was defined when a lock is performed. This event will be emitted when a lock is successfully registered in the RSK network.
+
+Topics: 
+- signature: `lock_btc(address indexed receiver, bytes32 btcTxHash, string senderBtcAddress, int256 amount)`
+- receiver: where the RBTC will be received
+
+Data:
+- btcTxHash: hash of the BTC transaction that produced the lock
+- senderBtcAddress: who performed the lock. The string representation of the BTC address, e.g. base58, bech32
+- amount: weis transferred to the recipient
+
+#### Update collections
+
+The existing event for update collections will only change its formatting.
+
+Topics:
+- signature: `update_collections(address sender)`
+
+Data:
+- sender: RSK address of the Federate who performed the call
+
+#### Release requested - _new event_
+
+For the Federation to send BTC to another address, the Bridge itself is responsible for creating the transaction.
+There are a number of reasons for this:
+- a user sends a transaction to the Bridge, with value and no data. This is when a user manually wants to convert their RBTC to BTC.
+- a user locked BTC, but their BTC address is not whitelisted, so the BTC are refunded.
+- a user locked BTC, but this transaction makes the Federation balance exceed the configured lockin cap. [see RSKIP-134](RSKIP134.md)
+- a retiring Federation migrates its funds to the active Federation.
+
+The Bridge will emmit an event reflecting the requested Release, once this release has been confirmed, and before the Federation members will start to provide their signatures.
+
+Topics:
+- signature: `release_btc_requested(address indexed sender, bytes32 indexed rskTxHash, bytes32 indexed btcTxHash, int256 amount)`
+- sender: who requested the release. If this event was emitted in the Federation migration the address will be the Bridge address.
+- rskTxHash: hash of the transaction that originated this release.
+
+    If this event was emitted during a user release, the transaction hash will be one of the users calling the Bridge (different to the executing transaction).
+  
+    If this event was emitted in the Federation migration, the transaction hash to be used will be the executing transaction.
+
+    Once the RSKIP gets implemented, there is a chance there will be events that won't have the rsktTxHash set because the storage won't have the data. In that case, the presence of rskTxHash with value 0x0 should be assumed as release that were requested before the activation of the corresponding Network Upgrade, but were confirmed after its activation.
+- btcTxHash: hash of the created Bitcoin transaction
+
+Data:
+- amount: weis transferred to the Bridge
+
+### Add signature
+
+This existing event will now link to the RSK transaction that originated the BTC transaction being signed.
+
+Topics:
+- signature: `add_signature(bytes32 indexed releaseRskTxHash, address indexed federatorRskAddress, bytes32 federatorBtcPublicKey)`
```

Comment:
`federatorBtcPublicKey` should be of type `bytes` instead of `bytes32` since btc public keys are longer than 32 bytes